### PR TITLE
CXX-1817 Rename get_utf8() to get_string() to make it more user-friendly

### PR DESCRIPTION
**Pull request by rayangler**
Most of the changes include switching most calls to `get_utf8()` with `get_string()`. 

There are instances where `BSONCXX_SUPPRESS_DEPRECATION_WARNINGS_BEGIN` (and its corresponding closing macro) were used on calls to `get_utf8()` (or `get_##name()`) in order to not trigger the deprecation warnings/errors on Evergreen.

### CODE REVIEW

File: src/bsoncxx/builder/core.cpp [hunk 1]
```diff
@@ -646,12 +647,15 @@ core& core::concatenate(const bsoncxx::document::view& view) {
 
 core& core::append(const bsoncxx::types::bson_value::view& value) {
     switch (static_cast<int>(value.type())) {
+        // CXX-1817; deprecation warning suppressed for get_utf8()
```
Comment (bazile-clyde) [inline src/bsoncxx/builder/core.cpp:650]: Nice!

**Review by bazile-clyde — APPROVED** [review]
Great work! LGTM!

File: src/bsoncxx/document/element.hpp [hunk 2]
```diff
@@ -147,11 +147,23 @@ class BSONCXX_API element {
...
+    /// @deprecated use document::element::get_string() instead.
+    ///
+    /// @throws bsoncxx::exception if this element is not a b_utf8.
+    ///
+    /// @return the element's value.
+    ///
+    BSONCXX_DEPRECATED types::b_utf8 get_utf8() const;
+
+    ///
+    /// Getter for elements of the b_utf8, or string, type. This function acts as a
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply src/bsoncxx/document/element.hpp:159]: After discussion, we'll do option 1, but as follow-up work in CXX-2109.

**Review by kevinAlbs — APPROVED** [review]
LGTM!

**Pull request merged**